bndtools.m2e/src/bndtools/m2e/MavenWorkspaceRepository.java

```diff
@@ -0,0 +1,291 @@
+import org.bndtools.api.ILogger;
+import org.bndtools.api.Logger;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.project.IMavenProjectChangedListener;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+import org.eclipse.m2e.core.project.IMavenProjectRegistry;
+import org.eclipse.m2e.core.project.MavenProjectChangedEvent;
+import org.osgi.resource.Capability;
+import org.osgi.resource.Requirement;
+import org.osgi.service.repository.Repository;
+
+import aQute.bnd.osgi.Domain;
+import aQute.bnd.osgi.repository.BaseRepository;
+import aQute.bnd.service.Refreshable;
+import aQute.bnd.service.RepositoryPlugin;
+import aQute.bnd.version.Version;
+import aQute.lib.collections.SortedList;
+
+public class MavenWorkspaceRepository extends BaseRepository implements Repository, RepositoryPlugin, Refreshable, IMavenProjectChangedListener {
+
+    private final static ILogger logger = Logger.getLogger(MavenWorkspaceRepository.class);
+
+    private final IMavenProjectRegistry mavenProjectRegistry = MavenPlugin.getMavenProjectRegistry();
+
+    private boolean inited = false;
+
+    private final Map<String,Entry<IMavenProjectFacade,MavenProject>> bsnMap = new HashMap<>();
+
+    @Override
+    public Map<Requirement,Collection<Capability>> findProviders(Collection< ? extends Requirement> requirements) {
+        return null;
+    }
+
+    @Override
+    public boolean refresh() throws Exception {
+        return false;
+    }
+
+    @Override
+    public File getRoot() throws Exception {
+        return null;
+    }
+
+    @Override
+    public PutResult put(InputStream stream, PutOptions options) throws Exception {
+        return null;
+    }
+
+    @Override
+    public File get(final String bsn, final Version version, Map<String,String> properties, final DownloadListener... listeners) throws Exception {
+        if (!inited) {
+            init();
+        }
+
+        final Entry<IMavenProjectFacade,MavenProject> entry = bsnMap.get(bsn);
+
+        if (entry == null)
+            return null;
+
+        final MavenProject mavenProject = entry.getValue();
+
+        // add the eclipse project that this comes from so we can look it up in the launch
+        // see bndtools.launch.BndDependencySourceContainer.createSourceContainers()
+        final String projectName = entry.getKey().getProject().getName();
+        properties.put("sourceProjectName", projectName);
+
+        final File bundleFile = new File(mavenProject.getBuild().getDirectory(), mavenProject.getBuild().getFinalName() + ".jar");
+
+        if (listeners == null || listeners.length == 0) {
+            return bundleFile;
```

Comment:
You cannot just return bundlefile here. You may need to run the java mojo to make it.